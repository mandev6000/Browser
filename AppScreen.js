/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TextInput,
  View,
  Text,
  StatusBar,
  Button,
  TouchableHighlight
} from 'react-native';
import { WebView } from 'react-native-webview';

import {
  Colors,
} from 'react-native/Libraries/NewAppScreen';

export default class AppScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: "https://www.google.com/",
      showWebView: false
    };
  }

  onButtonPress() {
    if (this.state.text === "") {
      alert("Please enter valid URL")
      return
    }
    this.setState({ showWebView: true })
  }

  onPressClose() {
    this.setState({ showWebView: false })
  }

  render() {
    return (
      <>
      <StatusBar barStyle="dark-content" />
        <SafeAreaView style={{ flex: 1}}>
        {
          this.state.showWebView ?  (
            <>
            <TouchableHighlight onPress={this.onPressClose.bind(this)}>
              <Text style={styles.footer}>‚ùå</Text>
            </TouchableHighlight>
            <WebView
              source={{uri: this.state.text}}
            />
            </>
            )
           :
            <View style={styles.sectionContainer}>
              <Text style={styles.sectionDescription}>
                Enter the Google Drive url to browse
              </Text>
  
              <TextInput
                style={{height: 40, marginVertical: 10}}
                placeholder="Enter URL here"
                onChangeText={text => this.setState({text})}
                value={this.state.text}
              />
              <Button
                onPress={this.onButtonPress.bind(this)}
                title="Browse"
              />
            </View>
        }
        </SafeAreaView>
      </>
    );
  }
}

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.dark,
  },
  footer: {
    color: Colors.black,
    fontSize: 20,
    fontWeight: '600',
    paddingRight: 12,
    textAlign: 'right',
  },
});