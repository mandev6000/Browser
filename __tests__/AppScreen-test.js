/**
 * @format
 */

import 'react-native';
import React from 'react';
import AppScreen from '../AppScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AppScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
