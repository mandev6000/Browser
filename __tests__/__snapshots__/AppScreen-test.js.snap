// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 32,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#444",
          "fontSize": 18,
          "fontWeight": "400",
          "marginTop": 8,
        }
      }
    >
      Enter the Google Drive url to browse
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter URL here"
      rejectResponderTermination={true}
      style={
        Object {
          "height": 40,
          "marginVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
      value="https://www.google.com/"
    />
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Browse
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
